lib/ldp.js

```diff
@@ -156,42 +154,40 @@ class LDP {
       return callback(error(500, "Can't parse container"))
     }
 
-    async.waterfall(
-      [
-        // add container stats
-        function (next) {
-          ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph, next)
-        },
-        // reading directory
-        function (next) {
-          ldpContainer.readdir(filename, next)
-        },
-        // Iterate through all the files
-        function (files, next) {
-          async.each(
-            files,
-            function (file, cb) {
-              let fileUri = url.resolve(reqUri, encodeURIComponent(file))
-              ldpContainer.addFile(ldp, resourceGraph, reqUri, fileUri, uri,
-                filename, file, cb)
-            },
-            next)
-        }
-      ],
-      function (err, data) {
+    // add container stats
+    new Promise((resolve, reject) =>
+      ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph,
+        err => err ? reject(err) : resolve())
+    )
+    // read directory
+    .then(() => new Promise((resolve, reject) =>
+      ldpContainer.readdir(filename,
+        (err, files) => err ? reject(err) : resolve(files))
```

Comment:
Good points!

> I was hesitant to do this here and in other places of the Solid code, because callbacks are sometimes used inconsistently (they don't always have an error argument).

Oof yeah that complicates things!

> And eventually, we should migrate to an await/async style, I suppose.

Sure thing. Though async/await is syntactic sugar on top of Promises, so I think that "promisifying" our interfaces goes hand-in-hand with async/await-ifying them :)

I like your point that this change is more of an interface redesign than simply "promisifying" the existing APIs. How about we simply leave things as they are here, and I'll create a ticket for making the async APIs more consistent.